### DISCUSSION TITLE
IoT Device tester compatibility check

### DISCUSSION THREAD

**Discussion by jainvikas8** [aws4embeddedlinux/meta-aws#7611] [Q&A]
For recipes-iot/aws-iot-greengrass/greengrass-bin_2.9.6.bb what would be the required IDT version, as we don't see this here - https://docs.aws.amazon.com/greengrass/v2/developerguide/dev-test-versions.html

```
$ sudo java -jar /greengrass/v2/alts/init/distro/lib/Greengrass.jar --version
AWS Greengrass v2.9.6
```

**Comment by nenadilic84** [comment]
IDT v4.9.1 for AWS IoT Greengrass
Supported AWS IoT Greengrass versions:
Greengrass nucleus v2.12.0, v2.11.0, v2.10.0, and v2.9.5

**Reply by jainvikas8** [reply]
Yes, but when running IDT v4.9.1 for the preinstalled version, it complains it's incompatible.  (2.9.6 != 2.9.5). Normally, it would make sense if IDT would support 2.9.x or 2.10.x etc

**Reply by jainvikas8** [reply]
Another problem I see is python3-pip is not installed as dependancy for `recipes-iot/aws-iot-greengrass/greengrass-bin_2.9.6.bb` , which suggests this version is not supported for testing. Is that the case?

**Comment by jainvikas8** [comment]
Any update on this please?

**Reply by thomas-roos** [reply]
Hi,
just did an update of the greengrass-bin recipe to version 2.12.5 - on *next branches - may this help?
Also regarding python3-pip - this could be installed, but there is no hard dependency on this.

**Reply by jainvikas8** [reply]
Will give it a try, but from past experiences, the IoT device tester fails the version check if it doesn't find a specific number.
